src/TakeoffController.cpp

```diff
@@ -0,0 +1,191 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Takeoff Controller
+//
+// Handles takeoff
+//
+////////////////////////////////////////////////////////////////////////////
+
+// Associated header
+#include "iarc7_motion/TakeoffController.hpp"
+
+// ROS Headers
+#include "ros_utils/ParamUtils.hpp"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.h"
+
+using namespace Iarc7Motion;
+
+TakeoffController::TakeoffController(
+        ros::NodeHandle& nh,
+        ros::NodeHandle& private_nh)
+    : transform_wrapper_(),
+      state_(TakeoffState::DONE),
+      throttle_(),
+      recorded_hover_throttle_(),
+      max_takeoff_start_height_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "max_takeoff_start_height")),
+      takeoff_complete_height_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "takeoff_complete_height")),
+      takeoff_throttle_ramp_rate_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "takeoff_throttle_ramp_rate")),
+      takeoff_throttle_bump_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "takeoff_throttle_bump")),
+      last_update_time_(),
+      startup_timeout_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "startup_timeout")),
+      update_timeout_(ros_utils::ParamUtils::getParam<double>(
+              private_nh,
+              "update_timeout"))
+{
+  // Saving this here to prevent compile warnings. It will be needed in a soon to come revision.
+  ros::NodeHandle nh_ = nh;
+}
+
+// Used to reset and check initial conditions for takeoff
+// Update needs to begin being called shortly after this is called.
+bool TakeoffController::resetForTakeover(const ros::Time& time)
+{
+    if (time < last_update_time_) {
+        ROS_ERROR("Tried to reset TakeoffHandler with time before last update");
+        return false;
+    }
+
+    // Get the current transform (xyz) of the quad
+    geometry_msgs::TransformStamped transform;
+    bool success = transform_wrapper_.getTransformAtTime(transform,
+                                                    "foot1",
```

Comment:
It's going away as soon as we have switches.

I'm not sure if this case has been tested, however this is the same order being used for quadvelocity controller. In addition it's working for the other checks.